internal/bminventory/inventory.go

```diff
@@ -4792,6 +4792,10 @@ func (b *bareMetalInventory) BindHostInternal(ctx context.Context, params instal
 	if err != nil {
 		return nil, common.NewApiError(http.StatusBadRequest, errors.Errorf("Failed to find cluster %s", params.BindHostParams.ClusterID))
 	}
+	allowed, err := b.authzHandler.HasAccessTo(ctx, cluster, auth.UpdateAction)
+	if err != nil || !allowed {
```

Comment:
It's enough to check !allowed because we return false in case of an error. This will make the code more readable. The error could be wrapped in application error here providing it can handle nil gracefully

Reply:
Fixed by checking 'allowed' only and added err to the log (as we don't want the expose the error message in the API response). We could consider logging only in the HasAccessTo func if it looks too cumbersome.